plugins/drop_folder/batch/Engine/KDropFolderFileTransferEngine.php

```diff
@@ -280,7 +280,19 @@ public static function getFileTransferManager(KalturaDropFolder $dropFolder)
 		if($dropFolder instanceof KalturaS3DropFolder)
 		{
 			$engineOptions['s3Region'] = $dropFolder->s3Region;
+
+			if ($dropFolder->useS3Arn)
+			{
+				$engineOptions['s3Arn'] = $dropFolder->s3Arn;
+				
+				if (empty($engineOptions['s3Arn']))
+				{
+					$msg = "Drop Folder ID [{$dropFolder->id}] enabled 'Bucket Policy Allow Access' but 's3Arn' value under 's3_drop_folder' in 'runtime_config' map is missing";
```

Comment:
Why no fail it in the step of the drop folder creation? Would be much more clean.